smallrye-reactive-messaging-rabbitmq/src/main/java/io/smallrye/reactive/messaging/rabbitmq/RabbitMQMessageSender.java

```diff
@@ -0,0 +1,272 @@
+                    } catch (Exception e) {
+                        // Message can't be sent - nacking and skipping.
+                        message.nack(e);
+                        RabbitMQLogging.log.serializationFailure(configuration.getChannel(), e);
+                        return Uni.createFrom().nullItem();
+                    }
+                })
+                .subscribe().with(
+                        tuple -> {
+                            if (tuple != null) { // Serialization issue
+                                subscriber.onNext(tuple.getItem2());
+
+                                if (inflights != Long.MAX_VALUE) {
+                                    upstream.get().request(1);
+                                }
+                            }
+                        },
+                        subscriber::onError);
+    }
+
+    /**
+     * Failed terminal state.
+     * <p>
+     * No further events will be sent even if {@link Subscription#request(long)} is invoked again.
+     *
+     * @param t the throwable signaled
+     */
+    @Override
+    public void onError(Throwable t) {
+        Subscription sub = upstream.getAndSet(Subscriptions.CANCELLED);
+        Subscriber<? super Message<?>> subscriber = this.downstream.get();
+        if (sub != null && sub != Subscriptions.CANCELLED && subscriber != null) {
+            subscriber.onError(t);
+        }
+    }
+
+    /**
+     * Successful terminal state.
+     * <p>
+     * No further events will be sent even if {@link Subscription#request(long)} is invoked again.
+     */
+    @Override
+    public void onComplete() {
+        Subscription sub = upstream.getAndSet(Subscriptions.CANCELLED);
+        Subscriber<? super Message<?>> subscriber = this.downstream.get();
+        if (sub != null && sub != Subscriptions.CANCELLED && subscriber != null) {
+            subscriber.onComplete();
+        }
+    }
+
+    /* ----------------------------------------------------- */
+    /* METHODS OF SUBSCRIPTION */
+    /* ----------------------------------------------------- */
+
+    /**
+     * No events will be sent by a {@link Publisher} until demand is signaled via this method.
+     * <p>
+     * It can be called however often and whenever needed—but if the outstanding cumulative demand ever becomes Long.MAX_VALUE
+     * or more,
+     * it may be treated by the {@link Publisher} as "effectively unbounded".
+     * <p>
+     * Whatever has been requested can be sent by the {@link Publisher} so only signal demand for what can be safely handled.
+     * <p>
+     * A {@link Publisher} can send less than is requested if the stream ends but
+     * then must emit either {@link Subscriber#onError(Throwable)} or {@link Subscriber#onComplete()}.
+     *
+     * @param l the strictly positive number of elements to requests to the upstream {@link Publisher}
+     */
+    @Override
+    public void request(long l) {
+        if (l != Long.MAX_VALUE) {
```

Comment:
This is not correct.
You need to keep track of the number of requests, add l and if the sum is greater that Long.MAX cap it to Long.MAX. 
Long.MAX means "unlimited" so, you won't need to decrease or keep track of the requests anymore.

request the new value if it was not Long.MAX before.

Note that we may need a way for `inflight` to indicate no-backpressure (even if not recommended)

Reply:
Ok, I got it now.
So this method is expected to be called only `once`.